### PR TITLE
Add actions for build and deploy workflows

### PR DESCRIPTION
**Pull request by Bonkles**
This PR creates two new actions, 'Build' (which fires only on PRs and reports any issues inline) and 'Deploy' (fires upon each commit, and uploads a build to the AWS bucket). 

The 'Build' action is super simple, and largely self explanatory.  Hopefully it will work as soon as I hit 'Create Pull Request'. 🤡 

_[Editor: Reader, it did not. ]_

There are a couple of interesting things going on with the 'Deploy' action, however. 

During build time, it must:
1) Parse the SHA of the git commit.
2) Create a destination path to the bucket. Since the bucket builds all go inside a `tileviewer` folder, the dest path is of the form `/tileviewer/<shorthash>-<nodeversion>-dist/`.
3) Provide the folder path from step 2 to the webpack invocation (done via environment variables), which in turn sets the  the 'CESIUM_BASE_URL' which the cesium library will use to fetch resources at runtime. 
4) Also modify the index.html `<script defer src=app.js>` tag to instead look for app.js within that same path. 

Each folder is uploaded to the bucket using the short git hash of the commit as a unique prefix- this busts any caching issues that might cause issues during testing and eliminates the 'refresh' problem during fast code iterations.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Deploy
+
+on:
+  push:
+    branches: [ master, develop, cicd, '*staging*' ]
```
Comment (Bonkles) [inline .github/workflows/deploy.yml:8]: Two things: 

1) We should rename 'master' to 'main'. 
2) We can delete the 'cicd' directive here once this branch has been merged.

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    - name: Download artifacts 📥
+      uses: actions/download-artifact@v2
+      with:
+        name: build-artifacts
+        path: dist
+
+    - name: Configure AWS credentials 🔐
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        role-to-assume: arn:aws:iam::632571768781:role/tileviewer-cicd-role
```
Comment (Bonkles) [inline .github/workflows/deploy.yml:65]: This is the exact ARN of the role I set up in IAM to allow read/write access to the tile viewer bucket.

File: webpack.config.js [hunk 3]
```diff
@@ -6,63 +6,70 @@ const CopyWebpackPlugin = require('copy-webpack-plugin');
...
+        patterns: [
+          { from: path.join(cesiumSource, cesiumWorkers), to: "Workers" },
+          { from: path.join(cesiumSource, "Assets"), to: "Assets" },
+          { from: path.join(cesiumSource, "Widgets"), to: "Widgets" },
+        ],
+      }),
+      new webpack.DefinePlugin({
+        // Define relative base path in cesium for loading assets
+        CESIUM_BASE_URL: JSON.stringify(
+          env.CI ? `/tileviewer/${env.DIST_BASE_URL}/` : ""
```
Comment (Bonkles) [inline webpack.config.js:69]: 'CI' is a variable that's only true while we're running in GH actions, the 'DIST_BASE_URL' is also fed in from the deploy.yml.

**Review by bsbates-fb — APPROVED** [review]

**Comment by Bonkles** [comment]
Retargeting this one to main didn't work because main is off in some funky state. I'll need to re-create the PR and re-target the master branch.

**Pull request closed without merging**